src/test/java/com/devonfw/sample/archunit/NamingConventionTest.java

```diff
@@ -51,6 +63,90 @@ public class NamingConventionTest {
                     .that().areAssignableTo(AbstractEto.class)
                     .should().haveSimpleNameEndingWith("Eto")
                     .because("Classes extending AbstractEto must follow the naming convention by ending with 'Eto'.");
+    /**
+     * DevonApplicationPersistenceEntityCheck verifying that classes extending ApplicationPersistenceEntity have to have be in layer dataaccess
+     */
+    @ArchTest
+    private static final ArchRule DevonApplicationPersistenceEntityCheck = 
+            classes()
+                   .that().areAssignableTo(ApplicationPersistenceEntity.class).should().resideInAnyPackage("..dataaccess..")
+                   .because("Classes extending ApplicationPersistenceEntity must be located in the package dataaccess");
+
+    /**
+     * DevonAbstractUcCheck verifying that classes extending AbstractUc have to have be in layer logic
+     */
+    @ArchTest
+    private static final ArchRule DevonAbstractUcCheck = 
+            classes()
+                   .that().areAssignableTo(AbstractUc.class).should().resideInAnyPackage("..logic..")
```

Comment:
`.andShould().haveSimpleNameStartingWith("Uc")`

ATTENTION: We do have some base-usecases starting with `AbstractUc` so either we add this with OR conjunction or even better we distinguish abstract from non-abstract classes here.